servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRoutes.java

```diff
@@ -173,9 +187,18 @@ protected GrpcService newServiceFromRoutes(final AllGrpcRoutes routes) {
     }
 
     @Nullable
-    private GrpcExecutionStrategy executionStrategy(final Method method, final Class<?> clazz) {
-        return getAndValidateRouteExecutionStrategyAnnotationIfPresent(method, clazz, strategyFactory, errors,
-                noOffloadsStrategy());
+    private GrpcExecutionStrategy executionStrategy(final String path, final Method method, final Class<?> clazz) {
+        // Check if we already have a computed GrpcExecutionStrategy for this path. This may happen when we re-register
+        // filtered routes:
+        final GrpcExecutionStrategy saved = executionStrategies.getOrDefault(path, NULL);
```

Comment:
Just use `get()` and eliminate defining `NULL` as default?

```java
final GrpcExecutionStrategy saved = executionStrategies.get(path);
if (saved != null) {
    return saved;
}
```

Reply:
Let's continue this discussion here: https://github.com/apple/servicetalk/pull/932#discussion_r374957744